From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chad Monroe <chad@monroe.io>
Date: Mon, 23 Mar 2026 15:57:14 -0700
Subject: [PATCH 948/957] wifi: mt76: fix stuck TX queues after SER with no
 active interfaces

When a firmware watchdog triggers full SER recovery before any VAPs
are up, mac80211 skips drv_reconfig_complete() because open_count is
zero. This leaves the DRIVER queue stop reason set permanently,
blocking all TX when interfaces eventually start.

Signed-off-by: Chad Monroe <chad@monroe.io>
Signed-off-by: Felix Fietkau <nbd@nbd.name>
(cherry picked from commit 130273a947ce7636ac1533e89b624061f8944fd1)
---
 mt7915/mac.c | 4 ++++
 mt7996/mac.c | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/mt7915/mac.c b/mt7915/mac.c
index 3d7d71ac..02c3735e 100644
--- a/mt7915/mac.c
+++ b/mt7915/mac.c
@@ -1464,6 +1464,10 @@ mt7915_mac_full_reset(struct mt7915_dev *dev)
 
 	mutex_unlock(&dev->mt76.mutex);
 
+	ieee80211_wake_queues(mt76_hw(dev));
+	if (ext_phy)
+		ieee80211_wake_queues(ext_phy->hw);
+
 	ieee80211_restart_hw(mt76_hw(dev));
 	if (ext_phy)
 		ieee80211_restart_hw(ext_phy->hw);
diff --git a/mt7996/mac.c b/mt7996/mac.c
index f3aa322b..73fce7f5 100644
--- a/mt7996/mac.c
+++ b/mt7996/mac.c
@@ -2511,6 +2511,8 @@ mt7996_mac_full_reset(struct mt7996_dev *dev)
 
 	mutex_unlock(&dev->mt76.mutex);
 
+	ieee80211_wake_queues(mt76_hw(dev));
+
 	ieee80211_restart_hw(mt76_hw(dev));
 }
 
